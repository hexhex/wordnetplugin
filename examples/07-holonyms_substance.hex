%% this returns all (substance) holonyms of a word
%% POS: nouns

%% the search number for (substance) holonyms
%% is 7.
%% Note: In the WordNet documentation this is
%%       the search number for substance meronyms.
%%       In fact search 7 fetches words, for
%%       which *word* is a meronym (and these
%%       are its holonyms)

word("hydrogen").
meronyms(W,SW,H,SH) :- &wordnet[07,W](1,SW,H,SH), word(W).

meronyms("hydrogen",1,"water",1)?
