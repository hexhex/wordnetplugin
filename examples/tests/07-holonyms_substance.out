{meronyms("hydrogen",1,"H2O",1), meronyms("hydrogen",1,"water",1), word("hydrogen")}
